### PR TITLE
Travis: Enforce PEP8

### PR DESCRIPTION
**Pull request by ax3l**
Add Python style checks to travis via flake8 tests.

### CODE REVIEW

File: docs/propose_changelog.py [hunk 1]
```diff
@@ -70,19 +72,20 @@
...
     pr_labels = i.labels
     pr_labels_names = map(lambda l: l.name, pr_labels)
+    pr_url = "https://github.com/ComputationalRadiationPhysics/picongpu/pull/"
     if "bug" in pr_labels_names:
-        if not "affects latest release" in pr_labels_names:
+        if "affects latest release" not in pr_labels_names:
             print("Filtering out development-only bug:")
             print("  #" + str(pr_nr) + " " + i.title)
-            print("  https://github.com/ComputationalRadiationPhysics/picongpu/pull/" + str(pr_nr))
+            print("  " + pr_url + str(pr_nr))
```
Comment (Heikman) [inline docs/propose_changelog.py]: FYI: `str` is not neccesary.

File: docs/propose_changelog.py [hunk 2]
```diff
@@ -70,19 +72,20 @@
...
             continue
 
     # filter out closed (unmerged) PRs
     pr = repo.get_pull(pr_nr)
     if not pr.merged:
         print("Filtering out unmerged PR:")
         print("  #" + str(pr_nr) + " " + i.title)
-        print("  https://github.com/ComputationalRadiationPhysics/picongpu/pull/" + str(pr_nr))
+        print("  " + pr_url + str(pr_nr))
```
Comment (Heikman) [inline docs/propose_changelog.py]: FYI: `str` is not neccesary.

File: src/tools/bin/plot_chargeConservation.py [hunk 3]
```diff
@@ -92,7 +91,8 @@ def plotError(h5file, slice_pos=[0.5, 0.5, 0.5]):
     for field_name in f["/data/{}/fields/".format(timestep)].keys():
         if field_name[-14:] == "_chargeDensity":
             # load species density
-            species_Density = np.array(f["/data/{}/fields/".format(timestep) + field_name])
+            species_Density = np.array(
+                f["/data/{}/fields/".format(timestep) + field_name])
```
Comment (Heikman) [inline src/tools/bin/plot_chargeConservation.py]: shift closing brace to a new line.

File: src/tools/bin/plot_chargeConservation_overTime.py [hunk 4]
```diff
@@ -189,36 +188,41 @@ def deviation_charge_conservation(h5file):
 
     ax1 = plt.subplot(121)
     ax1.set_xlabel(r"$t\,[\Delta t]$", fontsize=20)
-    ax1.set_ylabel(r"$\mathrm{max}|d|\,[\rho_\mathrm{max}(0)]$", fontsize=20)
+    ax1.set_ylabel(r"$\mathrm{max}|d|\,[\rho_\mathrm{max}(0)]$",
+                   fontsize=20)
```
Comment (Heikman) [inline src/tools/bin/plot_chargeConservation_overTime.py]: use standart indentation.

File: src/tools/bin/plot_chargeConservation_overTime.py [hunk 5]
```diff
@@ -189,36 +188,41 @@ def deviation_charge_conservation(h5file):
...
-    ax1.yaxis.set_major_locator( major_locator1 )
-    ax1.yaxis.set_major_formatter( major_formatter )
+    ax1.yaxis.set_major_locator(major_locator1)
+    ax1.yaxis.set_major_formatter(major_formatter)
 
     ax2 = plt.subplot(122)
     ax2.set_xlabel(r"$t\,[\Delta t]$", fontsize=20)
-    ax2.set_ylabel(r"$\left<|d|\right> \pm \sigma_d\,[\rho_\mathrm{max}(0)]$", fontsize=20)
+    ax2.set_ylabel(r"$\left<|d|\right> \pm \sigma_d\,[\rho_\mathrm{max}(0)]$",
+                   fontsize=20)
```
Comment (Heikman) [inline src/tools/bin/plot_chargeConservation_overTime.py]: use standart indentation.

File: src/tools/bin/smooth.py [hunk 6]
```diff
@@ -17,10 +17,8 @@
 # along with PIConGPU.
 # If not, see <http://www.gnu.org/licenses/>.
 #
-
-import numpy
-import sys
 from __future__ import division
+import numpy
```
Comment (Heikman) [inline src/tools/bin/smooth.py]: `import numpy as np`

File: src/tools/bin/plot_chargeConservation.py (hunk 3 above)
Comment (ax3l) [inline src/tools/bin/plot_chargeConservation.py]: why? it's not PEP8 enforced ;)

File: src/tools/bin/plot_chargeConservation_overTime.py (hunk 4 above)
Comment (ax3l) [inline src/tools/bin/plot_chargeConservation_overTime.py]: this is PEP8: doing 4 spaces would be a "under-indented block"

File: src/tools/bin/smooth.py (hunk 6 above)
Comment (ax3l) [inline src/tools/bin/smooth.py]: this will cause more changes but can do

File: src/tools/bin/plot_chargeConservation_overTime.py (hunk 4 above)
Comment (ax3l) [inline src/tools/bin/plot_chargeConservation_overTime.py]:
```python
    ax1.set_ylabel(r"$\mathrm{max}|d|\,[\rho_\mathrm{max}(0)]$",
        fontsize=20)
```
causes:
```
./src/tools/bin/plot_chargeConservation_overTime.py:192:9: E128 continuation line under-indented for visual indent
```

File: src/tools/bin/plot_chargeConservation_overTime.py (hunk 5 above)
Comment (ax3l) [inline src/tools/bin/plot_chargeConservation_overTime.py]: this is PEP8: doing 4 spaces would be a "under-indented block"

File: docs/propose_changelog.py (hunk 1 above)
Comment (ax3l) [inline docs/propose_changelog.py]: python2:
```
$ print(" aaa" + 3)
TypeError
----> 1 print(" aaa" + 3)

TypeError: cannot concatenate 'str' and 'int' objects
```

python3:
```
$ print(" aaa" + 3)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-1-cd690000aed7> in <module>()
----> 1 print(" aaa" + 3)

TypeError: Can't convert 'int' object to str implicitly
```

File: src/tools/bin/plot_chargeConservation_overTime.py (hunk 5 above)
Comment (ax3l) [inline src/tools/bin/plot_chargeConservation_overTime.py]: -> both on new line now

File: src/tools/bin/plot_chargeConservation_overTime.py (hunk 4 above)
Comment (ax3l) [inline src/tools/bin/plot_chargeConservation_overTime.py]: -> both on new line now

**Pull request merged**